core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -494,7 +495,15 @@ boolean addNode(ReaderNode node) {
 
         double lat = node.getLat();
         double lon = node.getLon();
-        double ele = eleProvider.getEle(lat, lon);
+        double ele = Double.NaN;
+        if (readElevationFromTag != null) {
+            String eleString = node.getTag(readElevationFromTag);
+            if (eleString != null)
+                ele = Double.parseDouble(eleString);
+        }
+        if (Double.isNaN(ele)) {
```

Comment:
I'm not really happy with this fallback mechanic. We might end up mixing values from different height models.
*If* we want to support this it's either tag based elevation or from an external provider.

Or to cite Ghostbuster: don't cross the streams